hawkbit-extension-artifact-repository-s3/src/main/java/org/eclipse/hawkbit/artifact/repository/S3Repository.java

```diff
@@ -131,37 +130,22 @@ private static String objectKey(final String tenant, final String sha1Hash) {
     @Override
     public AbstractDbArtifact getArtifactBySha1(final String tenant, final String sha1Hash) {
         final String key = objectKey(tenant, sha1Hash);
-
-        LOG.info("Retrieving S3 object from bucket {} and key {}", s3Properties.getBucketName(), key);
-        try (final S3Object s3Object = amazonS3.getObject(s3Properties.getBucketName(), key)) {
-            if (s3Object == null) {
-                return null;
-            }
-
-            final ObjectMetadata s3ObjectMetadata = s3Object.getObjectMetadata();
-
-            // the MD5Content is stored in the ETag
-            return new S3Artifact(amazonS3, s3Properties, key, sha1Hash,
-                    new DbArtifactHash(sha1Hash,
-                            BaseEncoding.base16().lowerCase().encode(
-                                    BaseEncoding.base64().decode(sanitizeEtag(s3ObjectMetadata.getETag()))),
-                            null),
-                    s3ObjectMetadata.getContentLength(), s3ObjectMetadata.getContentType());
-        } catch (final IOException e) {
-            LOG.error("Could not verify S3Object", e);
+        LOG.debug("Retrieving S3 object from bucket {} and key {}", s3Properties.getBucketName(), key);
+        try {
+            return S3Artifact.get(amazonS3, s3Properties, key, sha1Hash);
+        } catch (final S3ArtifactNotFoundException e) {
+            LOG.debug("Cannot find artifact for bucket {} with key {}", e.getBucket(), e.getKey(), e);
```

Comment:
here, however I would use level "warn" because we always execute `existsByTenantAndSha1` beforehand in usage scenarios